WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java

```diff
@@ -180,41 +181,43 @@ public void onBindViewHolder(final SiteViewHolder holder, int position) {
                 && !getItem(position + 1).isRecentPick;
         holder.divider.setVisibility(showDivider ?  View.VISIBLE : View.GONE);
 
-        holder.itemView.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                int clickedPosition = holder.getAdapterPosition();
-                if (isValidPosition(clickedPosition)) {
-                    if (mIsMultiSelectEnabled) {
-                        toggleSelection(clickedPosition);
-                    } else if (mSiteSelectedListener != null) {
-                        mSiteSelectedListener.onSiteClick(getItem(clickedPosition));
+        if (mIsMultiSelectEnabled || mSiteSelectedListener != null) {
```

Comment:
> but it could introduce the problem of the click listener not being assigned if the view is created before mSiteSelectedListener has been assigned.

[I added the notifyDataChanged](https://github.com/wordpress-mobile/WordPress-Android/commit/b7fcc74727d726aabd40c2b5fc6a2f6b0d29f004#diff-dc75c9e59c4d6b0cbad49a51353129f5R146) for that purpose basically. I tested it manually and verified that the ViewHolders get bound again but, please, give it a try on your end as well, thanks! For my test, I added a click listener to the whole layout to call `setOnSiteClickListener()` manually on click, long after the whole view has been set up.